src/util/k5test.py

```diff
@@ -241,7 +241,8 @@
   return code other than 0, expected_msg=MSG to expect a substring in
   the command output, and expected_trace=('a', 'b', ...) to expect an
   ordered series of line substrings in the command's KRB5_TRACE
-  output.
+  output, or tracefile=FILENAME to write trace output to the specified
```

Comment:
All uses of tracefile immediately open said file.  I'm wondering if it wouldn't be better to move that logic into `run` somehow - maybe a parameter to return the trace output instead of the command output, or perhaps interleave the two?

Not a huge issue though - if it's too complex to do that, this is fine as-is.

Reply:
I added a commit with my best idea along these lines, which is a "return_trace=True" parameter.  Let me know if it seems better or just awkward.  (This is the case I've run into where C's approach to multiple results is arguably more elegant than the functions-can-return-tuples approach taken by most modern languages.)  If we like it, I will either squash it into the first commit or make it a separate first commit.

Reply:
I think I slightly prefer it, but if you don't I'm fine with leaving it alone.